src/context.rs

```diff
@@ -63,6 +63,40 @@ impl Drop for RectangleList {
     }
 }
 
+impl fmt::Debug for RectangleList {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        if self.ptr.is_null() {
+            write!(f, "RectangleList")
+        } else {
+            unsafe {
+                let ptr = (*self.ptr).rectangles;
+                let len = (*self.ptr).num_rectangles;
+                let s = if ptr.is_null() || len == 0 {
+                    "[]".to_owned()
+                } else {
+                    format!("[{}]",
+                            slice::from_raw_parts(ptr, len as usize)
+                                .iter()
+                                .map(|s| format!("{:?}", s))
+                                .collect::<Vec<_>>()
+                                .join(", "))
+                };
+
+                write!(f, "RectangleList {{
```

Comment:
Always use the `Formatter::debug_*()` functions for such things, in this case `debug_list()` or even simpler

```rust
f.debug_tuple("RectangleList")
    .field(&*self)  // &* gives you a &[Rectangle]
    .finish()
```